srcJava/com/rti/perftest/ddsimpl/RTIDDSImpl.java

```diff
@@ -1088,15 +1081,14 @@ private void configureReaderQos(
             }
         }
 
-        if (_transport.allowsMulticast() && _isMulticast) {
-            String multicast_addr;
-
-            if (PerfTest.THROUGHPUT_TOPIC_NAME.equals(topicName)) {
-                multicast_addr = THROUGHPUT_MULTICAST_ADDR;
-            } else if (PerfTest.LATENCY_TOPIC_NAME.equals(topicName)) {
-                multicast_addr = LATENCY_MULTICAST_ADDR;
-            } else {
-                multicast_addr = ANNOUNCEMENT_MULTICAST_ADDR;
+        if (_transport.allowsMulticast()) {
+            String multicast_addr = _transport.getMulticastAddr(topicName);
+            if (multicast_addr == null) {
+                System.err.println("topic name must either be "
+                        + TopicName.THROUGHPUT + " or "
+                        + TopicName.LATENCY + " or "
+                        + TopicName.ANNOUNCEMENT);
+                return ;
```

Comment:
We have to improve the error handler:

Line 1007 if RTIDDSImpl.java
```
private bool configureReaderQos(
```
Line 1091 if RTIDDSImpl.java
```
return false;
```
Line 598 if RTIDDSImpl.java
```
if (!configureReaderQos(topicName, qosProfile, drQos)) {
    return null;
}
```
Line 1106 if RTIDDSImpl.java
```
return true;
```